inductiva/tasks/run_simulation.py

```diff
@@ -40,6 +40,9 @@ def run_simulation(
         "sim_dir": pathlib.Path,
     }
 
+    if computational_resources:
+        provider_id = computational_resources.provider
```

Comment:
I am a bit confused now. Provider id is linked to the resource (machine) but we can send a provider id on the simulator.run, that can be overwritten.

Another thing (correct me if I'm wrong) some machines might have no provider id right?  (as seen on the machine_groups.py)
```python
kwargs = {}
if is_elastic:
    mg_class = resources.ElasticMachineGroup
```

I think we need to clean this up, we should remove the provider_id from the simulator.run and make sure all machines have a provider_id.